api/buckets/service.go

```diff
@@ -73,33 +77,31 @@ func (s *Service) Init(ctx context.Context, req *pb.InitRequest) (*pb.InitReply,
 }
 
 func (s *Service) createBucket(ctx context.Context, dbID thread.ID, dbToken thread.Token, name string) (*Bucket, error) {
-	key, err := s.Collections.IPNSKeys.Create(ctx, dbID)
-	if err != nil {
-		return nil, err
-	}
-	bytes, err := crypto.MarshalPublicKey(key.Key.GetPublic())
-	if err != nil {
-		return nil, err
-	}
-	bkey, err := mbase.Encode(mbase.Base32, bytes)
-	if err != nil {
-		return nil, err
-	}
-
 	seed := make([]byte, 32)
 	if _, err := rand.Read(seed); err != nil {
 		return nil, err
 	}
 	pth, err := s.IPFSClient.Unixfs().Add(
 		ctx,
 		ipfsfiles.NewMapDirectory(map[string]ipfsfiles.Node{
-			bucketSeedName: ipfsfiles.NewBytesFile(seed),
+			seedName: ipfsfiles.NewBytesFile(seed),
 		}),
 		options.Unixfs.Pin(true))
 	if err != nil {
 		return nil, err
 	}
 
+	key, err := s.IPFSClient.Key().Generate(ctx, util.MakeToken(keyLen), opt.Key.Type(options.Ed25519Key))
```

Comment:
Use the key API to generate the key. We still need our own mapping to thread ID.